DataFormats/GEMRecHit/interface/ME0Segment.h

```diff
@@ -0,0 +1,85 @@
+#ifndef GEMRecHit_ME0Segment_h
+#define GEMRecHit_ME0Segment_h
+
+/** \class ME0Segment derived by the CSC segment
+ *  Describes a reconstructed track segment in the 6 layers of the ME0 system.
+ *  This is 4-dimensional since it has an origin (x,y) and a direction (x,y)
+ *  in the local coordinate system of the chamber.
+ *
+ *  $Date: 2014/02/04 12:41:32 $
+ *  \author Marcello Maggi
+ */
+
+#include <DataFormats/TrackingRecHit/interface/RecSegment.h>
+#include <DataFormats/GEMRecHit/interface/ME0RecHitCollection.h>
+
+#include <iosfwd>
+
+class ME0DetId;
+
+class ME0Segment GCC11_FINAL : public RecSegment {
+
+ public:
+
+  /// Default constructor
+ ME0Segment() : theChi2(0.){}
+  
+  /// Constructor
+  ME0Segment(const std::vector<const ME0RecHit*>& proto_segment, LocalPoint origin, 
+	     LocalVector direction, AlgebraicSymMatrix errors, double chi2);
+  
+  /// Destructor
+  virtual ~ME0Segment();
+
+  //--- Base class interface
+  ME0Segment* clone() const { return new ME0Segment(*this); }
+
+  LocalPoint localPosition() const { return theOrigin; }
```

Comment:
uhm, the return by value is from the base class interface (TrackingRecHit.h).
So, correction:
change to ``` virtual localPosition() const ```
same for all other inherited virtual methods